lib/client/cluster_client.go

```diff
@@ -312,6 +284,27 @@ func performMFACeremony(ctx context.Context, clt *ClusterClient, params ReissueP
 	if mfaChal == nil {
 		return nil, trace.BadParameter("server sent a %T on GenerateUserSingleUseCerts, expected MFAChallenge", resp.Response)
 	}
+
+	switch mfaChal.MFARequired {
+	case proto.MFARequired_MFA_REQUIRED_NO:
+		return nil, trace.Wrap(services.ErrSessionMFANotRequired)
+	case proto.MFARequired_MFA_REQUIRED_UNSPECIFIED:
+		if params.MFACheck != nil && !params.MFACheck.Required {
```

Comment:
In what situation would `params.MFACheck` be non-nil here? Is it trustworthy without a previous call to `IsMFARequired`?

(I genuinely don't know, so I assume a "casual" observer wouldn't either.)

Reply:
That's much clearer, thanks. I wish we had some of this background documented in the field itself.